BankManagerConsoleApp/BankManagerConsoleApp/BankManager.swift

```diff
@@ -5,40 +5,121 @@
 //  Created by 애쉬, 로빈 on 2022/11/04.
 //
 
+import Foundation
+
 struct BankManager {
-    private let bankWorker: BankWorkable
-    private let clientQueue: Queue<Client> = Queue()
+    private var bankWorkers: [BankWorker] = []
+    private let depositClientQueue: Queue<Client> = Queue()
+    private let loanClientQueue: Queue<Client> = Queue()
     private var totalClientCount: Int = 0
+    private var totalWorkTime: Double = 0.0
+    private let group: DispatchGroup = DispatchGroup()
+    
+    init(bankWorkers: [BankWorker]) {
+        self.bankWorkers = bankWorkers
+    }
     
-    init(bankWorker: BankWorkable) {
-        self.bankWorker = bankWorker
+    mutating func publishTicketNumber() -> Int {
+        let ticketNumber = totalClientCount + 1
+
+        totalClientCount = ticketNumber
+        
+        return ticketNumber
     }
     
     mutating func addClientQueue(_ client: Client) {
-        totalClientCount += 1
-        clientQueue.enqueue(client)
+        let requestingWork = client.requestingWork
+        
+        totalWorkTime += requestingWork.time
+        
+        switch requestingWork {
+        case .deposit:
+            depositClientQueue.enqueue(client)
+        case .loan:
+            loanClientQueue.enqueue(client)
+        case .none:
+            return
+        }
+    }
+    
+    mutating func open() {
+        designateWorkType()
+        assignBankWork()
     }
     
-    private func directBankWork() {
-        while !clientQueue.isEmpty {
-            guard let client = clientQueue.dequeue() else { return }
-            
-            bankWorker.startWork(for: client)
+    mutating private func designateWorkType() {
+        var currentDepositWorkerCount: Int = 0
+        
+        for index in 0..<bankWorkers.count {
```

Comment:
혀나블의 제안대로 `enumerated()`를 사용해 인덱스와 배열 요소에 한번에 접근해 수정하고자 하였습니다!
하지만, bankManager가 값타입이라 값을 복사해서만 가져오기 때문에 배열값의 프로퍼티를 수정할 수가 없었습니다 😭
또한, 배열의 값을 사용하지 않아 아래와 같은 에러 메시지가 발생했습니다.
>Unused Enumerated Violation: When the item is not used, `.indices` should be used instead of `.enumerated()`. (unused_enumerated)"

```swift
for (index, _) in bankWorkers.enumerated() {
      if currentDepositWorkerCount < DepositWorkerNumber.max {
                bankWorkers[index].bankWork = .deposit
                currentDepositWorkerCount += 1
      } else {
           bankWorkers[index].bankWork = .loan
      }
}
```

그래서 저희는 인덱스에만 접근하기 때문에 대신 `indices()`라는 메서드를 사용하는 것으로 수정해보았습니다.
```swift
for index in bankWorkers.indices {
      if currentDepositWorkerCount < DepositWorkerNumber.max {
            bankWorkers[index].bankWork = .deposit
            currentDepositWorkerCount += 1
       } else {
            bankWorkers[index].bankWork = .loan
       }
}
```